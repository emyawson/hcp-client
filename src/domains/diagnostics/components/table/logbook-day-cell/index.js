export { LogbookDayCell } from './logbook-day-cell.component';
