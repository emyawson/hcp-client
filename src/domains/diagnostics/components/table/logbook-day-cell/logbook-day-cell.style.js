import styled from 'styled-components';

import { colors } from 'src/domains/diagnostics/styles';

export const DayLinesList = styled.ul`
  list-style: none;
  padding: 0 0.5rem;
  margin: 0;
  text-align: left;
  color: ${props => (props.isWeekendDay ? colors.blue : colors.black)};
`;

DayLinesList.displayName = 'DayLinesList';
