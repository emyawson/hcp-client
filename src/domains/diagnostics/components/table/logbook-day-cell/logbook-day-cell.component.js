import React from 'react';

import { convertJSDateGMT } from 'src/domains/diagnostics/utils';
import { isWeekendDay } from 'src/domains/diagnostics/utils/time.util';

import { DayLinesList } from './logbook-day-cell.style';

import {
  EmptyRows,
  LogbookCellBlock,
  LogbookCellBlocksContainer,
  LogbookTableCellDiv,
} from '..';

export const LogbookDayCell = ({ dayLines, numberOfRows, highlight }) => (
  <LogbookTableCellDiv
    key={`${dayLines.join(', ')}`}
    width="6.7rem"
    borderRight={true}
    highlight={highlight}
  >
    <LogbookCellBlocksContainer>
      <LogbookCellBlock align="left" flex={1}>
        <DayLinesList
          isWeekendDay={isWeekendDay(
            convertJSDateGMT(new Date(dayLines.join(' '))),
          )}
        >
          {dayLines.map(dayLine => <li key={dayLine}>{dayLine}</li>)}
        </DayLinesList>
      </LogbookCellBlock>
    </LogbookCellBlocksContainer>
    <EmptyRows
      numberOfRows={numberOfRows - (dayLines.length - 1)}
      numberOfCells={1}
    />
  </LogbookTableCellDiv>
);
