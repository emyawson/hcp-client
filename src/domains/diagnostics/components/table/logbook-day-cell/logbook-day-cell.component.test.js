import { shallow } from 'enzyme';
import React from 'react';

import { LogbookDayCell } from './logbook-day-cell.component';

const mockProps = {
  dayLines: ['Wednesday,', 'Jan 17, 2017'],
  numberOfRows: 4,
};

describe('LogbookDayCell test suite', () => {
  test('it renders correctly', () => {
    const wrapper = shallow(<LogbookDayCell {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
