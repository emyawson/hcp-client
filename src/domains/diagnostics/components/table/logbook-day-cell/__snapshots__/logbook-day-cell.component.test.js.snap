// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogbookDayCell test suite it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dayLines={
      Array [
        "Wednesday,",
        "Jan 17, 2017",
      ]
    }
    numberOfRows={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "Wednesday,, Jan 17, 2017",
    "nodeType": "function",
    "props": Object {
      "borderRight": true,
      "children": Array [
        <LogbookCellBlocksContainer>
          <Unknown
            align="left"
            flex={1}
          >
            <DayLinesList
              isWeekendDay={false}
            >
              <li>
                Wednesday,
              </li>
              <li>
                Jan 17, 2017
              </li>
            </DayLinesList>
          </Unknown>
        </LogbookCellBlocksContainer>,
        <Unknown
          numberOfCells={1}
          numberOfRows={3}
        />,
      ],
      "highlight": undefined,
      "width": "6.7rem",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            align="left"
            flex={1}
          >
            <DayLinesList
              isWeekendDay={false}
            >
              <li>
                Wednesday,
              </li>
              <li>
                Jan 17, 2017
              </li>
            </DayLinesList>
          </Unknown>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "align": "left",
            "children": <DayLinesList
              isWeekendDay={false}
            >
              <li>
                Wednesday,
              </li>
              <li>
                Jan 17, 2017
              </li>
            </DayLinesList>,
            "flex": 1,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <li>
                  Wednesday,
                </li>,
                <li>
                  Jan 17, 2017
                </li>,
              ],
              "isWeekendDay": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "Wednesday,",
                "nodeType": "host",
                "props": Object {
                  "children": "Wednesday,",
                },
                "ref": null,
                "rendered": "Wednesday,",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "Jan 17, 2017",
                "nodeType": "host",
                "props": Object {
                  "children": "Jan 17, 2017",
                },
                "ref": null,
                "rendered": "Jan 17, 2017",
                "type": "li",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "numberOfCells": 1,
          "numberOfRows": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "Wednesday,, Jan 17, 2017",
      "nodeType": "function",
      "props": Object {
        "borderRight": true,
        "children": Array [
          <LogbookCellBlocksContainer>
            <Unknown
              align="left"
              flex={1}
            >
              <DayLinesList
                isWeekendDay={false}
              >
                <li>
                  Wednesday,
                </li>
                <li>
                  Jan 17, 2017
                </li>
              </DayLinesList>
            </Unknown>
          </LogbookCellBlocksContainer>,
          <Unknown
            numberOfCells={1}
            numberOfRows={3}
          />,
        ],
        "highlight": undefined,
        "width": "6.7rem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              align="left"
              flex={1}
            >
              <DayLinesList
                isWeekendDay={false}
              >
                <li>
                  Wednesday,
                </li>
                <li>
                  Jan 17, 2017
                </li>
              </DayLinesList>
            </Unknown>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "align": "left",
              "children": <DayLinesList
                isWeekendDay={false}
              >
                <li>
                  Wednesday,
                </li>
                <li>
                  Jan 17, 2017
                </li>
              </DayLinesList>,
              "flex": 1,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <li>
                    Wednesday,
                  </li>,
                  <li>
                    Jan 17, 2017
                  </li>,
                ],
                "isWeekendDay": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "Wednesday,",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wednesday,",
                  },
                  "ref": null,
                  "rendered": "Wednesday,",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "Jan 17, 2017",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Jan 17, 2017",
                  },
                  "ref": null,
                  "rendered": "Jan 17, 2017",
                  "type": "li",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "numberOfCells": 1,
            "numberOfRows": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
